addons/sale/models/sale_order.py

```diff
@@ -1898,3 +1898,11 @@ def _is_paid(self):
         """
         self.ensure_one()
         return self.currency_id.compare_amounts(self.amount_paid, self.amount_total) >= 0
+
+    def _message_get_suggested_recipients(self):
+        recipients = super(SaleOrder, self)._message_get_suggested_recipients()
+        for order in self:
+            if order.partner_id:
```

Comment:
Not sure, because you need to loop twice over the record, but I find it clearer
```suggestion
        for order in self.filtered('partner_id'):
```